bundles/org.openhab.binding.volvooncall/src/main/java/org/openhab/binding/volvooncall/internal/dto/Status.java

```diff
@@ -40,14 +40,33 @@
     public @NonNullByDefault({}) OnOffType engineRunning;
     public @NonNullByDefault({}) String brakeFluid;
     public @NonNullByDefault({}) String washerFluidLevel;
-    public @Nullable WindowsStatus windows;
-    public @Nullable DoorsStatus doors;
-    public @Nullable TyrePressure tyrePressure;
-    public @Nullable HvBattery hvBattery;
-    public @Nullable Heater heater;
-
+    private @NonNullByDefault({}) WindowsStatus windows;
```

Comment:
This seems to be wrong. The annotation says "I know for sure that whenever this field is accessed, it is non-null and that is assured by the framework". If that would be true, you don't need an `Optional` later on.

Reply:
I had to do this because if I put @Nullable instead, the compiler complains because the optional should be Optional<@Nullable WindowsStatus> and then everything behing goes wrong. But making another test, Eclipse complains but mvn builds it :-/

Reply:
I've changed the problem severity in eclipse, still shows as a warning, but as mvn does not complain, I suppose it's fine